relayer/relay.go

```diff
@@ -137,16 +134,14 @@ func (r *Relayer) StartBlockProcessors(ctx context.Context, errorChan chan error
 	var eg errgroup.Group
 
 	for _, chainRuntime := range r.chains {
-		chainRuntime := chainRuntime
```

Comment:
removal of this line is problematic, because all the go routines have the probability of referring `chainRuntime` to the last value of iteration of chains.
For example:
```go
func main() {
	values := []int{1, 2, 3, 4, 5}

	wg := &sync.WaitGroup{}
	for _, val := range values {
		wg.Add(1)
		go func() {
			defer wg.Done()
			fmt.Println("Value: ", val)
		}()
	}

	wg.Wait()
}
```
Running this code will not print all different values.

Reply:
It has been addressed on go 1.22.
https://go.dev/blog/loopvar-preview